dowhy/causal_refuters/sensitivity_analysis.py

```diff
@@ -0,0 +1,449 @@
+import statsmodels.formula.api as smfa
+import numpy as np
+from scipy.stats import t
+import pandas as pd
+import statsmodels.api as sma
+import logging
+import matplotlib.pyplot as plt
+
+
+class Sensitivity_analysis:
+    """
+    Class to perform sensitivity analysis
+    :param formula: The formula specifying the model
+    :param data: Pandas dataframe
+    :param treatment_name: name of treatment
+    :param q: percentage reduction for robustness value
+    :param standard_error: standard error in regression
+    :param estimate: estimate of regression
+    :param degree_of_freedom: degree of freedom of error in regression
+    :param h0: hypothesis
+    :param stats: dictionary for sensitivity statistics
+    :param benchmark_covariates: names of variables for benchmark bounding
+    :param kd: strength of association between benchmark and treatment variable to test with benchmark bounding
+    :param ky: strength of association between benchmark and outcome variable to test with benchmark bounding
+    :param r2dz_x: partial R^2  of putative unobserved confounder "z" with treatment "d", observed covariates "x" partialed out
+    :param r2yz_dx: partial R^2  of putative unobserved confounder "z" with outcome "y", observed covariates "x" and treatment "d" partialed out
+    :param r2dxj_x: partial R^2 of covariate Xj with treatment "d", covariates "x" excluding Xj are partialed out
+    :param r2yxj_dx:  partial R^2 of covariate Xj with outcome "y", covariates "x" excluding Xj and treatment "d" are partialed out
+    :param bounds_results: dataframe containing information about bounds and bias adjusted terms
+    :param stats: dictionary containing information like robustness value, partial R^2, estimate, standard error , degree of freedom, partial f^2, t-statistic
+    """
+    formula = None
+    data = None
+    treatment_name = None
+    estimate = None
+    degree_of_freedom = None
+    standard_error = None
+    q = None
+    confidence = None
+    h0 = 0
+    increase = False
+    benchmark_covariates = None
+    kd = None
+    ky = None
+    r2dz_x = None
+    r2yz_dx = None
+    r2dxj_x = None
+    r2yxj_dx = None
+    bias_adjusted_estimate = None
+    bias_adjusted_se = None
+    bias_adjusted_t = None
+    bias_adjusted_lower_CI = None
+    bias_adjusted_upper_CI = None
+    bounds_result = None
+    stats = None
+    
+
+    def __init__(self, formula = None, data = None, treatment_name = None, q = 1.0, confidence = 0.05, increase = False, benchmark_covariates = None, kd = None, ky = None):

```

Comment:
I'd suggest removing formula, data and treatment_name from the parameter list.

formula can be derived from identified_estimand. 
each refuter object has a `self._data` corrresponding to dataset.
And treatment_name is also provided to each refuter as self._treatment_name